packages/webclient/src/Authorizer/AzureCLIAuthorizer.ts

```diff
@@ -0,0 +1,99 @@
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as io from '@actions/io';
+import { IAuthorizer } from "./IAuthorizer";
+
+export class AzureCLIAuthorizer implements IAuthorizer{
+    private constructor() { }
+
+    public static async getAuthorizer() {
+        if(!this._authorizer) {            
+            this._authorizer = new AzureCLIAuthorizer();
+            await this._authorizer._initialize();
+        }
+
+        return this._authorizer;
+    }
+
+    public get subscriptionID(): string {
+        return this._subscriptionId;
+    }
+
+    public get baseUrl(): string {
+        return this._cloudEndpoints['resourceManager'] || 'https://management.azure.com/';
+    }
+
+    public getCloudSuffixUrl(suffixName: string): string {
+        return this._cloudSuffixes[suffixName];
+    }
+
+    public getCloudEndpointUrl(endpointName: string): string {
+        return this._cloudEndpoints[endpointName];
+    }
+
+    public async getToken(force?: boolean, args?: string[]): Promise<string> {
+        if(!this._token || force) {  
+            try {
+                let azAccessToken = JSON.parse(await AzureCLIAuthorizer.executeAzCliCommand('account get-access-token', !!args ? args : []));
+                console.log(`::add-mask::${azAccessToken}`);
```

Comment:
instead use core.setSecret() it is recommended by actions team, maybe because they might change implementation later